### PR TITLE
Fixing the innodb stats when multiple bufferpools exist

### PR DESCRIPTION
**Pull request by art-spilgames**
This fixes the innodb stats when multiple bufferpools exist. Currently they are adding up or overwriting the totals and averages with the latest individual bufferpool statistics as the output is parsed sequentially.
By chunking the output first and ignoring the individual bufferpool statistics this should be fixed for now.

Todo for the future: allow the individual statistics to be sent alongside the totals/averages

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -117,6 +120,110 @@ Coding standards
...
+-  mysql (connecitons, queries, etc)
+-  metrics (metrics to be stored including their type)
+
+Daemon
+------
+The daemon section allows you to set the paths to your log and pic files
+
+Statsd
+------
+The Stasd section allows you to configure the prefix and hostname of the
```
Comment (spil-jasper) [inline README.rst]: Typo: Stasd -> Statsd

File: README.rst [hunk 2]
```diff
@@ -117,6 +120,110 @@ Coding standards
...
+The Stasd section allows you to configure the prefix and hostname of the 
+metrics. In our example the prefix has been set to mysql and the hostname 
+is included. This will log the status.com_select metric to:
+mysql.<hostname>.status.com_select
+
+You can use any prefix that is necessary in your environment.
+
+MySQL
+-----
+The MySQL section allow you to configure the credentials of your mysql host
```
Comment (spil-jasper) [inline README.rst]: allow -> allows

File: README.rst [hunk 3]
```diff
@@ -117,6 +120,110 @@ Coding standards
...
+of your own application data inside your application database you *could* 
+create your own custom query this way. So for example:
+::
+    stats_types = myapp
+    query_myapp = SELECT some_metric_name, some_metric_value FROM myapp.metric_table WHERE metric_ts >= DATE_SUB(NOW(), interval 1 MINUTE)
+    interval_myapp = 60000
+
+This will query your application database every 60 seconds, fetch all the 
+metrics that have changed since then and send them through StatsD.
+Obviously you need to whitelist them via the metric section below.
```
Comment (spil-jasper) [inline README.rst]: metric -> metrics

File: README.rst [hunk 4]
```diff
@@ -117,6 +120,110 @@ Coding standards
...
+Gauges are sticky values (like the spedometer in your car). Rates are the 
+number of units that need to be translated to units per second. Timers are 
+the time it took to perform a certain task.
+
+An ever increasing value like the com\_select can be sent various ways. If you 
+wish to retain the absolute value of the com_select it is advised to configure 
+it as a gauge. However if you are going to use it as a rate (queries per 
+second) it is no use storing it as a rate in the first place and then later 
+on calculate the integral of the gauge to get the rate. It would be far more 
+acurate to store it as a rate in the first place.
```
Comment (spil-jasper) [inline README.rst]: acurate -> accurate

File: mysql_statsd/thread_mysql.py [hunk 5]
```diff
@@ -90,12 +91,24 @@ def _run(self):
...
                 rows = self._preprocess(check_type, column_names, cursor.fetchall())
                 for key, value in rows:
-                    metric_key = check_type+"."+key.lower()
-                    metric_type = self.metrics.get(metric_key)
-
-                    # Only allow the whitelisted metrics to be sent off to Statsd
-                    if metric_key in self.metrics:
-                        self.queue.put((metric_key, value, metric_type))
+                    key = key.lower()
+                    metric_key = check_type+"."+key
```
Comment (spil-jasper) [inline mysql_statsd/thread_mysql.py]: Please use spaces between operators, like so:
x = a + "foo" + b

File: mysql_statsd/thread_mysql.py [hunk 6]
```diff
@@ -90,12 +91,24 @@ def _run(self):
...
+
+                    # Support multiple bufferpools in metrics (or rather: ignore them)
+                    # Bascially bufferpool_* whitelists metrics for *all* bufferpools
+                    if key.startswith('bufferpool_'):
+                        # Rewrite the metric key to the whitelisted wildcard key
+                        whitelist_key = check_type+"."+re.sub(r'(.*bufferpool_)\d+(\..+)', r'\1*\2', key)
```
Comment (spil-jasper) [inline mysql_statsd/thread_mysql.py]: Nicer would be:
```whitelist_key = "{0}.{1}.format(re.sub( ... ),
                                                   check_type)
```

**Comment by spil-jasper** [comment]
Great job on the documentation! I commented on some typos and code-style issues in-line.

For making the commit history in this project nice, could you also squash your commits into a single one? You can use `git rebase -i` to do that.

File: mysql_statsd/thread_statsd.py [hunk 7]
```diff
@@ -51,17 +51,17 @@ def send_stat(self, item):
             sender = self.get_sender(t)
             sender(k, float(delta))
         else:
-           sender = self.get_sender(t)
-           sender(k, float(v))
+            sender = self.get_sender(t)
```
Comment (thijsdezoete) [inline mysql_statsd/thread_statsd.py]: These spaces are out of line

**Comment by thijsdezoete** [comment]
Looking good. Could you please fix the indentation for us? It seems there are a couple of indentation differences in the files already

**Pull request merged**